### ISSUE TITLE
Consider using a "comment-system as a service"

### ISSUE THREAD

**Issue by bmintz** [r3bl/blog.r3bl.me#5]
I use Disqus on my blog and it works great. It's proprietary though. Someday I'll make my own comment-system as a service system...

**Comment by aleksandar-todorovic** [comment]
Hi,

I've been actually using Disqus for years on this website. I've only recently decided to shut it down, mostly for the reasons discussed here: https://notes.ayushsharma.in/2017/09/im-killing-disqus-comments-on-my-blog-heres-why

**tl;dr:** It affects the website speed, makes way too many third-party requests, and adds a section that I don't control, affecting my reader's privacy.

In fact, the code is still there, and the only thing that's stopping the comments section from appearing is that I explicitly set `comments: false` in YAML header of every article.

Looking at the Disqus dashboard, I've received 1.4 comments per article for years. Considering all the downsides, I've decided to remove it as an option.

With that said, I've already looked at the alternatives, and Mozilla's [@coralproject/talk](https://github.com/coralproject/talk) seems to tick all the marks of what I want from a commenting service.

The downside of it:

> Talk is a Node application with dependencies managed by Yarn that connects to MongoDB and Redis databases in order to persist data.

So yeah, I would love to include it on my blog, but I still haven't played around with incorporating its technology stack with a static site generator.

**Bottom line:** "Commenting as a Service" is a no for privacy reasons, but I'm looking at self-hosted alternatives.

**Comment by aleksandar-todorovic** [comment]
I should point out that I have a similar opinion about an analytics service as well. I'm currently using [Clicky Pro](https://clicky.com/help/pricing), whose terms and conditions don't allow them to sell the data about my visitors.

However, it's still a third-party service that I would love to get rid of the moment I can dedicate some time to spin up an instance of Piwik or some alternative. Sooner or later, my web projects will become entirely self-hosted. Unfortunately, I'm not there yet.

**Comment by bmintz** [comment]
I was mostly suggesting Comments, yeah. I'm wondering if maybe you could use something which simply takes in comments from a static form and commits them to a separate repo… then the static HTML built by that webhook would be included as a submodule from this repo.

**Comment by bmintz** [comment]
Message edits and deletions would not be possible this way, however. It would be purely manual/PR based.

The upshot of that approach is that the only dynamic part is the webhook. Everything else in that system is static.
